common/txmgr/confirmer.go

```diff
@@ -326,7 +335,20 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pro
 
 	if ec.resumeCallback != nil {
 		mark = time.Now()
-		if err := ec.ResumePendingTaskRuns(ctx, head); err != nil {
+
+		// TODO update this after BCI-3573 and BCI-3730 is merged
+		var latestFinalizedHeadNum int64
+		if !ec.chainConfig.FinalityTagEnabled() {
+			latestFinalizedHeadNum = head.BlockNumber()
+		} else if _, latestFinalizedBlock, err := ec.headTracker.LatestAndFinalizedBlock(ctx); err != nil {
+			return fmt.Errorf("failed to retrieve latest finalized head: %w", err)
+		} else {
+			latestFinalizedHeadNum = latestFinalizedBlock.BlockNumber()
+		}
+
+		// TODO Once BCI-3574 is merged we can update the chainConfig interface to remove FinalityTagEnabled, and remove the
+		//  head.BlockNumber()
+		if err := ec.ResumePendingTaskRuns(ctx, latestFinalizedHeadNum); err != nil {
```

Comment:
I wonder if we even have to deal with the latestFinalizedHeadNum here. Once the Finalizer PR is merged, I think we could just update the query inside this method to check transaction state for `finalized` instead of checking block ranges? Taking an even further step back, maybe this is logic that could move to the Finalizer since it deals with finality. Feel like that could be a better separation of responsibility but maybe out of scope for this change.